src/trio/_sync.py

```diff
@@ -112,6 +112,12 @@ def statistics(self) -> EventStatistics:
         """
         return EventStatistics(tasks_waiting=len(self._tasks))
 
+    def __bool__(self) -> NoReturn:
+        """Raise NotImplementedError."""
+        raise NotImplementedError(
+            "Trio events cannot be treated as bools; consider using 'event.is_set()'"
+        )
```

Comment:
I think we should warn that this is deprecated and return `True` for now?

Reply:
Changed to this in 1ea6615